qiime2/metadata.py

```diff
@@ -91,6 +102,69 @@ def load(cls, path):
             raise ValueError(invalid_metadata_template % msg)
         return cls(df)
 
+    def merge(self, *others):
+        """Merge this ``Metadata`` object with other ``Metadata`` objects.
+
+        Returns a new ``Metadata`` object containing the merged contents of
+        this ``Metadata`` object and `others`. The merge is not in-place and
+        will always return a **new** merged ``Metadata`` object.
+
+        The merge will include only those IDs that are shared across **all**
+        ``Metadata`` objects being merged (i.e. the merge is an *inner join*).
+
+        Each column being merged must be unique; merging metadata with
+        overlapping columns will result in an error.
+
+        Parameters
+        ----------
+        others : tuple
+            Zero or more ``Metadata`` objects to merge with this ``Metadata``
+            object.
+
+        Returns
+        -------
+        Metadata
+            New object containing merged metadata. The merged IDs will be in
+            the same relative order as the IDs in this ``Metadata`` object
+            after performing the inner join. The merged column order will
+            match the column order of ``Metadata`` objects being merged
+            from left to right.
+
+        Notes
+        -----
+        The merged metadata object tracks all source artifacts that it was
+        built from to preserve provenance (i.e. the ``.artifacts`` property
+        on all ``Metadata`` objects is merged).
+
+        """
+        dfs = []
+        columns = []
+        artifacts = []
+        for md in itertools.chain([self], others):
+            df = md._dataframe
+            dfs.append(df)
+            columns.extend(df.columns.tolist())
+            artifacts.extend(md.artifacts)
+
+        columns = pd.Index(columns)
+        if columns.has_duplicates:
+            raise ValueError(
+                "Cannot merge metadata with overlapping columns. The "
+                "following columns overlap: %s" %
+                ', '.join([repr(e) for e in columns.get_duplicates()]))
+
+        merged_df = dfs[0].join(dfs[1:], how='inner')
```

Comment:
Since the first artifact defines the index order, having this be a method makes a little more sense to me now.